platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/cache/CookieCache.java

```diff
@@ -124,32 +128,79 @@ public synchronized void setExpirationTime(int expirationMinutes) {
     }
 
     /**
-     * Listens for removal notifications from the cache and logs each time a removal is performed.
+     * Check to see if the provided Date/Time is outside of the SAML assertion time bounds
+     *
+     * @param key         the corresponding key for the assertion
+     * @param checkedDate the Date object to compare against SAML assertion time bounds
+     * @return true if the current time is outside of the SAML assertion time bounds
      */
-    private class RemovalListenerLogger implements RemovalListener<String, DataWrapper> {
-
-        @Override
-        public void onRemoval(RemovalNotification<String, DataWrapper> notification) {
-            LOGGER.debug("Expiring SAML ref:assertion {}:{} due to {}.",
-                    notification.getKey(),
-                    notification.getValue(),
-                    notification.getCause()
-                            .toString());
+    public boolean checkIfOutsideOfSSOTimeBounds(String key, Date checkedDate) {
+        Element element = getSamlAssertion(key);
+
+        if (element != null) {
+            SecurityToken securityToken = new SecurityToken();
+            securityToken.setToken(element);
+            SecurityAssertionImpl assertion = new SecurityAssertionImpl(securityToken);
+
+            if (checkedDate.before(assertion.getNotBefore())) {
+                LOGGER.debug("SAML Assertion Time Bound Check Failed.");
+                LOGGER.debug("\t Checked time of {} is before the NotBefore time of {}",
+                        checkedDate,
+                        assertion.getNotBefore());
+                return true;
+            }
+            if (checkedDate.equals(assertion.getNotOnOrAfter())
+                    || checkedDate.after(assertion.getNotOnOrAfter())) {
+                LOGGER.debug("SAML Assertion Time Bound Check Failed.");
+                LOGGER.debug(
+                        "\t Checked time of {} is equal to or after the NotOnOrAfter time of {}",
+                        checkedDate,
+                        assertion.getNotOnOrAfter());
+                return true;
+            }
         }
+
+        return false;
     }
 
-    private static class DataWrapper {
-        private Element element;
+    /**
+     * Check to see if the current Date/Time is outside of the SAML assertion time bounds
+     *
+     * @param key the corresponding key for the assertion
+     * @return true if the current time is outside of the SAML assertion time bounds
+     */
+    public boolean checkIfOutsideOfSSOTimeBounds(String key) {
+        // Call with new Date, which gives us the current time (now)
```

Comment:
Well the story behind it is that I originally built the method with the checkedDate arg thinking about testability... I was planning on passing in different dates to check, but then I realized that this was not the true use case here, that in reality we would most probably be checking the SAMLAssertion condition dates against "now". I decided to leave it overloaded because of the potential versatility, but if it adds confusion and people ask me to remove it then I can certainly do so :)